### PR TITLE
Add search box

### PR DESCRIPTION
**Pull request by fanxhe-pair**
Issue Search #42

### CODE REVIEW

**Comment by hagenburger** [comment]
1. `.lsg--sub-headline` is missing in the search scope.
2. Currently it displays the matching headline only. It would be great, if the example (or any other content) until the next headline (or end of document) would also be shown.

**Comment by fanxhe-pair** [comment]
@hagenburger we couldn't assign you

File: stylesheets/livingstyleguide/_search.scss [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#search {
```
Comment (hagenburger) [inline stylesheets/livingstyleguide/_search.scss]: For consistency, I would prefer `.lsg--search` over `#search`.

File: stylesheets/livingstyleguide/_search.scss [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+  margin: 0px auto;
+  width:  $lsg--width + ($lsg--width / 2);
```
Comment (hagenburger) [inline stylesheets/livingstyleguide/_search.scss]: Remove one space after `width:`

File: stylesheets/livingstyleguide/_search.scss [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+}
+
+#search-box{
```
Comment (hagenburger) [inline stylesheets/livingstyleguide/_search.scss]: For consistency, I would prefer `.lsg--search-box` over `#search-box`.

File: lib/livingstyleguide/templates/search-box.html.erb [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    if (node.nodeType !== Node.TEXT_NODE) {
+      if (node.className.includes("lsg--")) {
+        matched.push(node)
+      }
+    }
+  }
+
+  var addNextElement = function(selection, matched, i) {
+    var j = i + 1;
+    while(j < selection.length) {
```
Comment (hagenburger) [inline lib/livingstyleguide/templates/search-box.html.erb]: Add a space after `while`.

File: lib/livingstyleguide/templates/search-box.html.erb [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+      addMatch(matched, selection[j]);
+      j++;
+    }
+  }
+
+  var addPreviousElement = function(selection, matched, i) {
+    var j = i;
+    while(j >= 0) {
+      addMatch(matched, selection[j]);
+      if(hasClass(selection[j], "lsg--headline")) {
```
Comment (hagenburger) [inline lib/livingstyleguide/templates/search-box.html.erb]: Add a space after `if`.

File: lib/livingstyleguide/templates/search-box.html.erb [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+        break;
+      }
+      addMatch(matched, selection[j]);
+      j++;
+    }
+  }
+
+  var addPreviousElement = function(selection, matched, i) {
+    var j = i;
+    while(j >= 0) {
```
Comment (hagenburger) [inline lib/livingstyleguide/templates/search-box.html.erb]: Add a space after `while`.

File: stylesheets/livingstyleguide/_search.scss [hunk 7]
```diff
@@ -0,0 +1,10 @@
...
+  bottom: 2 * $lsg--gap-width;
```
Comment (hagenburger) [inline stylesheets/livingstyleguide/_search.scss]: Prefer using `margin` over `position: relative`.

**Comment by fanxhe-pair** [comment]
Ping @hagenburger we fixed coding style and added a search-box filter :smile: !!

**Pull request merged**